[GEN 2][Mobile] Add the missing mobile pieces for deployment over mobile. (#7186)
aws-amplify/docs: src/pages/[platform]/deploy-and-host/fullstack-branching/branch-deployments/index.mdx

  <source src="/images/gen2/fullstack-branching/autobranch.mp4" />
</video>

3. Push a commit to your `feature/A` and `staging` branches that match the pattern. You should start seeing deployments on the console page. You will now have three fullstack branches deployed.

![Production, feature/A, and staging branches listed on app overview page in Amplify console.](/images/gen2/fullstack-branching/branches.png)

## Promote changes to production

In Gen 2, promoting changes to production follows the normal Git-based workflow.

![Workflow for merging changes from feature/A branch to main, or production, branch.](/images/gen2/fullstack-branching/gitflow.png)

1. Make a change in your `feature/A` branch.

> ```sh showLineNumbers={false}

Reviewer: ````suggestion
```bash title="Terminal" showLineNumbers={false}
````

git checkout -b feature/A

## make some edits to your code

git commit --am "New data model to track comments for todos added"

git push origin feature/A
```

2. Submit a pull request to your `main` branch. Once your team has validated the changes, merge the pull request to `main`. This will initiate a build on your `main` branch and update any frontend or backend resources that you changed.

### Generate client config

You can generate the config for a branch environment by running:

<InlineFilter filters={['angular','javascript','nextjs','react','react-native','swift','vue']}>
For Web and React Native, generating the config with the default format and output directory.

> ```bash showLineNumbers={false}

Reviewer: ````suggestion
```bash title="Terminal" showLineNumbers={false}
````

npx amplify generate config --app-id <your-amplify-app-id> --branch main --out-dir path/to/config
```
</InlineFilter>

<InlineFilter filters={['android']}>
<Callout warning>
  Be sure to add a "raw" folder under app/src/main/res directory if it doesn't
  exist.
</Callout>

> ```bash showLineNumbers={false}

Reviewer: ````suggestion
```bash title="Terminal" showLineNumbers={false}
````

npx amplify generate config --app-id=<app-id> --branch main --format json-mobile --out-dir app/src/main/res/raw
```
</InlineFilter>
<InlineFilter filters={['flutter']}>

> ```bash showLineNumbers={false}

Reviewer: ````suggestion
```bash title="Terminal" showLineNumbers={false}
````

</video>

3. Push a commit to your `feature/A` and `staging` branches that match the pattern. You should start seeing deployments on the console page. You will now have three fullstack branches deployed.

![Production, feature/A, and staging branches listed on app overview page in Amplify console.](/images/gen2/fullstack-branching/branches.png)

## Promote changes to production

In Gen 2, promoting changes to production follows the normal Git-based workflow.

![Workflow for merging changes from feature/A branch to main, or production, branch.](/images/gen2/fullstack-branching/gitflow.png)

1. Make a change in your `feature/A` branch.

```sh showLineNumbers={false}
git checkout -b feature/A

## make some edits to your code

git commit --am "New data model to track comments for todos added"

git push origin feature/A
```

2. Submit a pull request to your `main` branch. Once your team has validated the changes, merge the pull request to `main`. This will initiate a build on your `main` branch and update any frontend or backend resources that you changed.

### Generate client config

You can generate the config for a branch environment by running:

```bash showLineNumbers={false}

> npx amplify generate config --app-id <AMPLIFY_APP_ID> --branch main --format=["mjs", "json", "json-mobile", "ts", "dart"] --out-dir=path/to/config

Reviewer: ```suggestion
npx amplify generate config --app-id <your-amplify-app-id> --branch main --out-dir path/to/config
```

nit for placeholder text https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md#16-bracket-replacement-text-in-codeblocks

also format will soon be just `json` and `dart`, and the equal sign is optional, but for consistency we can remove them

```

For Web and React Native, generating the config with the default format and output directory.

<BlockSwitcher>
<Block name="Flutter">
```bash showLineNumbers={false}

> npx amplify generate config --app-id=<app-id> --branch=main --format=dart --out-dir=lib

Reviewer: ```suggestion
npx amplify generate config --app-id <your-amplify-app-id> --branch main --format dart --out-dir lib
```
